### PR TITLE
Add clock message support

### PR DESCRIPTION
**Pull request by Shokman**

### CODE REVIEW

**Comment by caguero** [comment]
+1

**Comment by iche033** [comment]
Do we need to add the functions to `convert_builtin_interfaces.hpp`? This seems to compile fine without them. Maybe @caguero  know about this?

**Comment by Shokman** [comment]
Is it going to work without at least the ign to ros? The ros to ign is unnecessary but I added it just in case... It is a bridge so I thought it would be incomplete without it.

**Comment by iche033** [comment]
I was thinking more of the function declarations in `convert_builtin_interfaces.hpp`. Currently the new functions that exist in .cpp file are not added to the .hpp file, i.e.:

```
template<>
void
convert_ign_to_1(
  const ignition::msgs::Clock & ign_msg,
  rosgraph_msgs::Clock & ros1_msg);

template<>
void
convert_1_to_ign(
  const rosgraph_msgs::Clock & ros1_msg,
  ignition::msgs::Clock & ign_msg);
```

So not sure if need to add them or not

File: test/test_utils.h [hunk 1]
```diff
@@ -173,6 +174,25 @@ namespace testing
...
+    _msg.clock.sec  = 1;
+    _msg.clock.nsec = 2;
+  }
+
+  /// \brief Compare a message with the populated for testing.
+  /// \param[in] _msg The message to compare.
+  void compareTestMsg(const rosgraph_msgs::Clock &_msg)
+  {
+    rosgraph_msgs::Clock expected_msg;
+    createTestMsg(expected_msg);
```
Comment (iche033) [inline test/test_utils.h]: `expected_msg` is not being used in this function?

**Comment by iche033** [comment]
ping @Shokman @caguero 

do we want to get this in for the release?

**Comment by Shokman** [comment]
I think we should. Clock message is important for many messages. I am updating this PR

**Comment by caguero** [comment]
> Do we need to add the functions to `convert_builtin_interfaces.hpp`? This seems to compile fine without them. Maybe @caguero know more about this?

Yes please. Add them to `convert_builtin_interfaces.hpp` and to `builtim_interfaces_factories.hpp`.

**Comment by caguero** [comment]
+1 to merge this pull request before the next release.

File: test/test_utils.h (hunk 1 above)
Comment (Shokman) [reply test/test_utils.h]: fixed

**Comment by Shokman** [comment]
I updated the Pr and addressed the comments.

**Comment by caguero** [comment]
Could you also add the new declarations to `convert_builtin_interfaces.hpp` and to `builtin_interfaces_factories.hpp` please?

**Pull request merged**